src/main.rs

```diff
@@ -167,10 +323,17 @@ mod test {
         let (_in_file, in_string) = read_file(Path::new(input)).unwrap();
         let (_out_file, out_string) = read_file(Path::new(&output)).unwrap();
 
-        let lines: Vec<&str> = in_string.split("\n").collect();
+        let lines: Vec<Line> = in_string
+            .split("\n")
+            .map(|line| Line {
+                should_format: true,
```

Comment:
Just to get the tests working for the initial draft of this PR, I'm essentially running them in `--full-spec` mode, but as mentioned in the OP I should probably include tests that exercise the real git diff scoping mechanism.